etna/packages/etna-js/utils/fetch.jsx

```diff
@@ -34,7 +34,7 @@ export const headers = (...types) => {
 export const json_fetch = (method) => (path, params) =>
   fetch(path, {
     method,
-    credentials: 'same-origin',
+    credentials: 'include',
```

Comment:
Is there a good reason for this? The same-origin policy is much safer.

Reply:
Without CORS credentials, Timur can't make a request to Metis to upload things...

Reply:
Note that the `magma_api` request also uses CORS credentials. The alternative is to not share the upload components  or inject the fetch utility or something....seems messier.

Reply:
The same-origin policy ought to work for metis as it does for magma, as long as the domain (e.g. ucsf.edu) is the same.

Reply:
So if I return that to `same-origin`, Metis tries to call Janus to authenticate the user, and rejects the request. Note that [when Timur calls Magma](https://github.com/mountetna/monoetna/blob/master/timur/lib/client/jsx/api/magma_api.js#L12), it also has to use `include` for `credentials`, not  `same-origin` -- do you have an example of where Timur calls Magma using `same-origin` policy and it works?

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://janus-dev.etna-development.org/login?refer=http%3A%2…339c187cd380a16c6273%26X-Etna-Id%3Dmagma%26X-Etna-Headers%3D. (Reason: CORS request did not succeed).
```

<img width="877" alt="Screen Shot 2020-06-25 at 10 15 49 AM" src="https://user-images.githubusercontent.com/4930129/85736909-6e773800-b6ee-11ea-8f53-4db6fee68e60.png">

Reply:
> The same-origin policy ought to work for metis as it does for magma, as long as the domain (e.g. ucsf.edu) is the same.

Yeah, as far as I know this isn't true.  SOP is in consideration of the entire FQN, including subdomain. https://en.wikipedia.org/wiki/Same-origin_policy#Origin_determination_rules is a good workflow to check

Reply:
You're right, this is not the case; we relax it on the server end using Etna::CrossOrigin, but the client doesn't relax it right now.

We can instead apparently [relax the same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) in the client by setting e.g. `document.domain = 'ucsf.edu'`. I'm not sure this is actually a significant improvement but in these matters I'd prefer to be as conservative as possible.

Reply:
Interesting, I didn't know you could actually alter the domain directly.

Reply:
Relevant issue I just wrote up which might obviate some of this discussion: https://github.com/mountetna/etna/issues/60

Reply:
Modifying `document.domain` seems like a pretty neat idea, but doesn't seem to work in practice (Like Zach, that's new to me). I wonder if because the APIs are still in subdomains and aren't on the top-level domain.

I set Timur to use `document.domain = document.domain.split('.').slice(1).join('.');`, which for local dev for me becomes `etna-development.org`. And then I changed the `magma_api` calls to use `credentials: same-origin` instead of the current `includes`, and now I get a `401` Unauthorized. So I think there are still issues authenticating across services, where credentials have to be included.

Reply:
The alternative could be to read values from cookies and set them on requests explicitly as headers?
Then we don't have to worry SOP as it applies to cookies.

Reply:
Yeah, there's no reason we can't do this already in advance of mountetna/etna#60 - the auth layer already respects headers, the client is able to read the token in JS. Doesn't have to be immediately, though.

Reply:
:+1: